language/move-lang/src/test_utils/mod.rs

```diff
@@ -74,17 +75,31 @@ pub fn error(s: String) -> datatest_stable::Result<()> {
 pub fn ir_tests() -> impl Iterator<Item = (String, String)> {
     macro_rules! comp_to_string {
         ($comp_opt:expr) => {{
-            $comp_opt.unwrap().as_os_str().to_str()?.to_owned()
+            $comp_opt.as_os_str().to_str()?
         }};
     }
-    datatest_stable::utils::iterate_directory(Path::new(PATH_TO_IR_TESTS)).flat_map(|path| {
+    let num_components = Path::new(PATH_TO_IR_TESTS)
+        .canonicalize()
+        .unwrap()
+        .components()
+        .map(|_| 1)
+        .sum();
+    datatest_stable::utils::iterate_directory(Path::new(PATH_TO_IR_TESTS)).flat_map(move |path| {
         if path.extension()?.to_str()? != IR_EXTENSION {
             return None;
         }
         let pathbuf = path.canonicalize().ok()?;
-        let mut components = pathbuf.components().rev();
-        let name = comp_to_string!(components.next());
-        let dir = comp_to_string!(components.next());
+        let mut components = pathbuf.components();
```

Comment:
A comment would be helpful somewhere in here.... I looked at this for 2-3 minutes and I'm still not completely sure what this is doing.